provider/inspec.rb

```diff
@@ -73,8 +74,81 @@ def generate_typed_array(data, prop) end
 
     def emit_resourceref_object(data) end
 
-    def emit_nested_object(data) end
-
     def generate_network_datas(data, object) end
+
+    def emit_nested_object(data)
+      target = if data[:emit_array]
+                 data[:property].item_type.property_file
+               else
+                 data[:property].property_file
+               end
+      {
+        source: File.join('templates', 'inspec', 'nested_object.erb'),
+        target: "libraries/#{target}.rb",
+        overrides: emit_nested_object_overrides(data)
+      }
+    end
+
+    def emit_nested_object_overrides(data)
+      data.clone.merge(
+        api_name: data[:api_name].camelize(:upper),
+        object_type: data[:obj_name].camelize(:upper),
+        product_ns: data[:product_name].camelize(:upper),
+        class_name: if data[:emit_array]
+                      data[:property].item_type.property_class.last
+                    else
+                      data[:property].property_class.last
+                    end
+      )
+    end
+
+    def primitive?(property)
+      is_primitive = property.is_a?(::Api::Type::Primitive)
+      is_primitive_array = (property.is_a?(Api::Type::Array)\
+        && !property.item_type.is_a?(::Api::Type::NestedObject))
+      is_primitive || is_primitive_array
+    end
+
+    def resource_ref?(property)
```

Comment:
Does this need to be a separate function?

Reply:
We have been doing `.is_a?(Api::Type::Whatever)` throughout TF + Ansible, so I'm inclined to have you do that.

I'll let @SirGitsalot have the final say